OrbitCore/LinuxPerfEventProcessor.cpp

```diff
@@ -39,8 +41,12 @@ void LinuxPerfEventProcessor::ProcessAll()
 void LinuxPerfEventProcessor::ProcessTillOffset()
 {
     // Process the events in the event_queue
+    // We do not lock this read, as otherwise, 
+    //  it would lock the complete exceution of 
+    //  that function.
```

Comment:
This is not safe.  inline bool IsQueueEmpty() { ScopeLock lock(m_Mutex); return m_EventQueue.empty(); } would not lock the entire function.  In any case, aren't you ending up locking the bulk of the function anyway since there is a lock in the while loop?